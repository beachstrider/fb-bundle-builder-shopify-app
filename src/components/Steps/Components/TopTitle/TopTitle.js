import React from 'react'
import PropTypes from 'prop-types'
import styles from './TopTitle.module.scss'

const TopTitle = ({
  title,
  subTitle,
  showSeparator = true,
  className,
  children
}) => {
  return (
    <div className={`${styles.top} ${className}`}>
      <h2 className={`${styles.topTitle}`}>{title}</h2>
      <p className={`${styles.topSubtitle}`}>{subTitle}</p>
      {children && <p className={`${styles.topExtra}`}>{children}</p>}
      {showSeparator && (
        <div data-testid="separator" className={`${styles.topSeparator}`} />
      )}
    </div>
  )
}

TopTitle.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  subTitle: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  showSeparator: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node])
}

export default TopTitle
