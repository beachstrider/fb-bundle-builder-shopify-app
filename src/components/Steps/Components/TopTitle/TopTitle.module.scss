@import '../../../../variables';

$rowTopTitle: 494px;

.top {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  padding-bottom: 2rem;
  margin: auto;

  &Title {
    font-size: $fontLarge;
    font-family: $fontFamilyBlack !important;
    margin-top: 1.5rem;
    margin-bottom: 0;

    @media only screen and (max-width: $breakpointTablet) {
      font-size: $semiMediumFont;
    }
  }

  &Subtitle {
    font-size: $fontMediumRegular;
    font-family: $fontFamilyBook !important;
    width: $rowTopTitle;
    margin-top: 1rem;
    margin-bottom: 0.5rem;

    @media only screen and (max-width: $rowTopTitle) {
      width: $rowWidthMobile;
      padding: 0 2rem;
    }
  }

  &BoldSubtile {
    font-size: 2.2rem;
  }

  &Extra {
    font-size: $fontMediumSmall;
    font-family: $fontFamilyBook;
    margin: 0;
  }

  &Separator {
    display: none;
    margin-top: 1rem;
    width: 60px;
    border: 1px solid $border;

    @media only screen and (max-width: $breakpointTablet) {
      display: block !important;
    }
  }
}
