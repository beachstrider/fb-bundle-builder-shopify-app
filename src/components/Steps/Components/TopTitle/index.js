import TopTitle from './TopTitle'
export default TopTitle
