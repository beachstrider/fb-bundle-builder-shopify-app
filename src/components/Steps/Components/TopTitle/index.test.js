import React from 'react'
import { render, act, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import TopTitle from './TopTitle'

const renderComponent = (props) => {
  return render(<TopTitle {...props} />)
}

const defaultProps = {
  title: 'Title test',
  subTitle: 'Subtitle test'
}

describe('TopTitle component test', () => {
  describe('test render components props', () => {
    test('should render title and subtitle', () => {
      renderComponent(defaultProps)

      expect(screen.getByText('Title test')).toBeInTheDocument()
      expect(screen.getByText('Subtitle test')).toBeInTheDocument()
    })

    test('should render extra element', () => {
      const extraElement = 'This is an extra element'
      renderComponent({
        ...defaultProps,
        children: <span>{extraElement}</span>
      })

      expect(screen.getByText('Title test')).toBeInTheDocument()
      expect(screen.getByText('Subtitle test')).toBeInTheDocument()
      expect(screen.getByText(extraElement)).toBeInTheDocument()
    })

    test('should render separator', () => {
      renderComponent({
        ...defaultProps
      })

      expect(screen.getByText('Title test')).toBeInTheDocument()
      expect(screen.getByText('Subtitle test')).toBeInTheDocument()
      expect(screen.getByTestId('separator')).toBeVisible()
      expect(screen.getByTestId('separator')).toBeInTheDocument()
    })

    test('should render without separator', () => {
      renderComponent({
        ...defaultProps,
        showSeparator: false
      })

      expect(screen.getByText('Title test')).toBeInTheDocument()
      expect(screen.getByText('Subtitle test')).toBeInTheDocument()
      expect(screen.queryByTestId('separator')).not.toBeInTheDocument()
    })
  })

  describe('Accesibility test', () => {
    test('should test jest axe', async () => {
      await act(async () => {
        const { container } = renderComponent(defaultProps)

        expect(await axe(container)).toHaveNoViolations()
      })
    })
  })

  describe('regression test', () => {
    test('basic render', () => {
      const { asFragment } = renderComponent(defaultProps)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
