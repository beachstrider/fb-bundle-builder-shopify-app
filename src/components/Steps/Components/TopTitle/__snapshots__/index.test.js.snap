// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTitle component test regression test basic render 1`] = `
<DocumentFragment>
  <div
    class="top undefined"
  >
    <h2
      class="topTitle"
    >
      Title test
    </h2>
    <p
      class="topSubtitle"
    >
      Subtitle test
    </p>
    <div
      class="topSeparator"
      data-testid="separator"
    />
  </div>
</DocumentFragment>
`;
